src/services/azureFunctionsService.ts

```diff
@@ -73,11 +73,19 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 				LocalizedConstants.createProject, LocalizedConstants.learnMore);
 			if (projectCreate === LocalizedConstants.learnMore) {
 				vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(constants.sqlBindingsDoc));
+				return;
 			} else if (projectCreate === LocalizedConstants.createProject) {
 				// start the create azure function project flow
 				await azureFunctionApi.createFunction({});
 			}
-			return;
+			if (await azureFunctionUtils.getHostFiles()) {
+				// once project is created then start sql binding flow
+				projectFile = await azureFunctionUtils.getAzureFunctionProject();
+			} else {
+				// if user cancels prompt show warning
```

Comment:
Is this comment correct? The if has nothing to do about prompting the user for anything - it's just looking for the host files. And why do you check the host files first? That's already done as part of the `getAzureFunctionProject` flow so doing it here too seems unnecessary

Reply:
Oh I see - if they cancel out of the dialog on line 83 you're saying?

I'd suggest just calling `getAzureFunctionProject` instead - that already checks for host files and if none exist returns undefined.

Reply:
Line 79 specifically. `await azureFunctionApi.createFunction({});` - if user cancels that prompt early then we show the error. 

If the complete it then we do the getAzureFunctionProject call. But I cant just do getAzureFunctionProject call as I found if a user completes the createFunction promise then that is completed before the project is even made. Similar to what happened below in the logic with the `waitForNewFunctionFile`

Reply:
I don't think this is the right solution then - the check you added isn't waiting for anything to be created. It just immediately checks if the host.json file isn't there. But if the project isn't done being created we can't guarantee the host.json file is there.

So if you have seen that happening then I would suggest you look into adding a file watcher here for the host.json file being created. That's going to be a much more robust way of doing that. 

Although the tricky thing here is that we have no idea where the project is created, and IIRC the createFunction flow asks the user if they want to add it to their workspace so we aren't even guaranteed that the project will be visible to us after creation. I don't have a good answer for those right now, but we definitely should be doing something more than just hoping the host.json file is created by the time we call that if the createFunction call is returning before the project is completely created.